gateway/src/main/java/io/zeebe/gateway/ClusterClient.java

```diff
@@ -27,8 +32,34 @@ public ClusterClient(final ZeebeClient client) {
     this.client = client;
   }
 
-  public ActorFuture<Topology> sendRequest(final HealthRequest healthRequest) {
-    final ActorFuture<Topology> future = client.newTopologyRequest().send();
-    return future;
+  public ActorFuture<Topology> sendHealthRequest(final HealthRequest healthRequest) {
+    return client.newTopologyRequest().send();
+  }
+
+  public ActorFuture<DeploymentEvent> sendDeployWorkflowRequest(
+      final DeployWorkflowRequest deployRequest) {
+
+    if (deployRequest.getWorkflowsList().size() == 0) {
+      throw new RuntimeException("no workflow to deploy");
+    }
+
+    final List<WorkflowRequestObject> workflowsList = deployRequest.getWorkflowsList();
+    final int size = workflowsList.size();
+
+    WorkflowRequestObject cursor = workflowsList.get(size - 1);
+    DeployWorkflowCommandBuilderStep2 clusterRequestStep2 =
+        client
+            .workflowClient()
+            .newDeployCommand()
+            .addResourceBytes(cursor.getDefinition().toByteArray(), cursor.getName());
+
+    for (int i = size - 2; i >= 0; --i) {
```

Comment:
Why we need to iterate backwards?

Reply:
Cause writing obscure code matters. No, just kidding! I was too lazy to introduce another variable and reluctant to do too much math in the code so I just took the size and went backward. I guess we can rewrite this in a cleaner way. Any suggestions?